src/commands/runFromPackageDeploy.ts

```diff
@@ -0,0 +1,198 @@
+                kind: [],
+                learnMoreLink: 'https://aka.ms/T5o0nf'
+            });
+            if (sa.id) {
+                await updateWorkspaceSetting(settingKey, sa.id, fsPath);
+            }
+            const accountKey: string = 'AccountKey=';
+            name = sa.name;
+            key = sa.connectionString.substring(sa.connectionString.indexOf(accountKey) + accountKey.length);
+        }
+    } else {
+        const sClient: StorageClient = new StorageClient(node.credentials, node.subscriptionId);
+        const rg: string = azUtil.getResourceGroupFromId(storageAccountId);
+        name = azUtil.getNameFromId(storageAccountId);
+        const result: StorageAccountListKeysResult = await sClient.storageAccounts.listKeys(rg, name);
+        if (!result.keys || result.keys.length === 0) {
+            throw new ArgumentError(result);
+        } else {
+            key = result.keys[0].value;
+        }
+    }
+
+    if (name !== undefined && key !== undefined) {
+        return azureStorage.createBlobService(name, key);
+    } else {
+        throw new ArgumentError({ name: name, key: key });
+    }
+}
+
+async function createBlobFromZip(blobService: azureStorage.BlobService, zipFilePath: string, blobName: string): Promise<string> {
+    const containerName: string = 'azureappservice-run-from-zip';
+    await new Promise<void>((resolve: () => void, reject: (err: Error) => void): void => {
+        blobService.createContainerIfNotExists(containerName, (err: Error) => {
+            if (err !== null) {
+                reject(err);
+            } else {
+                resolve();
+            }
+        });
+    });
+
+    await new Promise<void>((resolve: () => void, reject: (err: Error) => void): void => {
+        blobService.createBlockBlobFromLocalFile(containerName, blobName, zipFilePath, (error: Error, _result: azureStorage.BlobService.BlobResult, _response: azureStorage.ServiceResponse) => {
+            if (error !== null) {
+                reject(error);
+
+            } else {
+                resolve();
+            }
+        });
+    });
+    const sasToken: string = blobService.generateSharedAccessSignature(containerName, blobName, <azureStorage.common.SharedAccessPolicy>{
+        AccessPolicy: {
+            Permissions: azureStorage.BlobUtilities.SharedAccessPermissions.READ + azureStorage.BlobUtilities.SharedAccessPermissions.LIST,
+            Start: azureStorage.date.secondsFromNow(-10),
+            // for clock desync
+            Expiry: azureStorage.date.daysFromNow(365),
+            ResourceTypes: azureStorage.BlobUtilities.BlobContainerPublicAccessType.BLOB
+        }
+    });
+
+    return blobService.getUrl(containerName, blobName, sasToken, true);
+}
+
+async function zipDirectory(fsPath: string, zipFileName?: string, globPattern: string = '**/*', ignorePattern?: string | string[]): Promise<{ zipFilePath: string, createdZip: boolean }> {
```

Comment:
You copied this from the shared package, right? That makes me think `runFromPackageDeploy` would fit better in the shared package

Reply:
I think that we discussed this back when I was implementing this.  If I recall correctly, we didn't want it in the shared package because no other extension would need `runFromPackageDeploy`.  Should we expose the `zipDirectory` function instead?

Reply:
I guess you said `func pack` zips for us, so I guess this comment thread no longer applies

Reply:
Well, we only run it for Python so we actually still need it for node.

Reply:
I guess I don't care too much. It seems wrong to expose `zipDirectory` from the "app service" package, so I'd rather move the deploy code over to the shared package or leave the copied zip code here as you have it.

Also definitely focus on Python. I don't care too much if JavaScript doesn't work.